.github/workflows/push-docker-on-release.yml

```diff
@@ -0,0 +1,50 @@
+name: Build and Push Docker Image
+
+on:
+  release:
+    types: [published]
```

Comment:
Building of binaries is kicked off from a tag, rather than release.  Should we do that here as well?
https://github.com/crate-ci/typos/blob/0d9e0c2c1bd7f770f6eb90f87780848ca02fc12c/.github/workflows/post-release.yml#L16-L21